### PR TITLE
Monitor Ingestion Troubleshooting Guide

### PR DESCRIPTION
**Pull request by gracewilcox**
Closes https://github.com/Azure/azure-sdk-for-go/issues/21257

Adds troubleshooting guide for the `azingest` module.

### CODE REVIEW

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+### Error Handling
+
+All methods which send HTTP requests return `*azcore.ResponseError` when these requests fail. `ResponseError` has error details and the raw response from Monitor Ingestion.
+
+### Logging
+
+To troubleshoot issues with the module, first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+This module uses the logging implementation in `azcore`. To turn on logging for all Azure SDK modules, set `AZURE_SDK_GO_LOGGING` to `all`. By default, the logger writes to stderr. Use the `azcore/log` package to control log output. For example, logging only HTTP request and response events, and printing them to stdout:
```
Comment (RickWinter) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:25]: Could call the specific log flags which would be most helpful to debug monitor issues.

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+// Includes only requests and responses in credential logs
+azlog.SetEvents(azlog.EventRequest, azlog.EventResponse)
+```
+
+## Troubleshooting logs ingestion
+
+### Troubleshooting authorization errors
+
+If you get an error with HTTP status code 403 (Forbidden), it means that the provided credentials have insufficient permissions to upload logs to the specified Data Collection Endpoint (DCE) and Data Collection Rule (DCR) ID.
```
Comment (RickWinter) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:43]:
```suggestion
If you get an error with HTTP status code 403 (Forbidden), it means the provided credentials lack sufficient permissions to upload logs to the specified Data Collection Endpoint (DCE) and Data Collection Rule (DCR) ID.
```

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+```text
+"error": {
+"code": "OperationFailed",
+"message": "The authentication token provided does not have access to ingest data for the data collection rule with immutable Id '***' PipelineAccessResult: AccessGranted: False, IsRbacPresent: False, IsDcrDceBindingValid: , DcrArmId: /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr, Message: Required autorization action was not found for tenantId *** objectId *** on resourceId /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr ConfigurationId: ***.."
```
Comment (RickWinter) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:48]:
```suggestion
"message": "The authentication token provided does not have access to ingest data for the data collection rule with immutable Id '***' PipelineAccessResult: AccessGranted: False, IsRbacPresent: False, IsDcrDceBindingValid: , DcrArmId: /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr, Message: Required authorization action was not found for tenantId *** objectId *** on resourceId /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr ConfigurationId: ***.."
```
Hopefully that type is just in this document.  If not please make a bug and fix the SDK or worse if the message comes from the service file a bug on the service.

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+   * To ingest logs, ensure the service principal is assigned the **Monitoring Metrics Publisher** role for the DCR.
+1. If the user or application is granted sufficient privileges to upload logs, ensure you're authenticating as that user/application. If you're authenticating using the [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/azidentity#defaultazurecredential), check the logs to verify that the credential used is the one you expected. To enable logging, see the [Enable client logging](#logging) section.
+1. The permissions may take up to 30 minutes to propagate. If the permissions were granted recently, retry after some time.
+
+### Troubleshooting missing logs
+
+When you send logs to Azure Monitor for ingestion, the request may succeed, but you may not see the data appear in the designated Log Analytics workspace table as configured in the DCR. To investigate and resolve this issue, ensure the following:
+
+* Double-check that you're using the correct DCE when configuring the `LogsIngestionClient`. Using the wrong endpoint can result in data not being properly sent to the Log Analytics workspace.
```
Comment (RickWinter) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:62]: Is DCR and DCE common acronyms in monitor?  Perhaps write it out in long form

**Review by RickWinter — APPROVED** [review]

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    * [Error Handling](#error-handling)
+    * [Logging](#logging)
+* [Troubleshooting logs ingestion](#troubleshooting-logs-ingestion)
+    * [Troubleshooting authorization errors](#troubleshooting-authorization-errors)
+    * [Troubleshooting missing logs](#troubleshooting-missing-logs)
+    * [Troubleshooting slow logs upload](#troubleshooting-slow-logs-upload)
+
+## General troubleshooting
+
+### Error Handling
```
Comment (scottaddie) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:17]:
```suggestion
### Error handling
```

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 1 above)
Comment (scottaddie) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:25]:
```suggestion
This module uses the logging implementation in `azcore`. To turn on logging for all Azure SDK modules, set `AZURE_SDK_GO_LOGGING` to `all`. By default, the logger writes to `stderr`. Use the `azcore/log` package to control log output. For example, logging only HTTP request and response events, and printing them to `stdout`:
```

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 4 above)
Comment (scottaddie) [reply sdk/monitor/azingest/TROUBLESHOOTING.md:62]: They're already defined in the "Troubleshooting authorization errors" section, so we're good to go

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+* Keep in mind that data ingestion may take some time, especially if you're sending data to a specific table for the first time. In such cases, allow up to 15 minutes for the data to be fully ingested and available for querying and analysis.
+
+### Troubleshooting slow logs upload
+
+If you experience delays when uploading logs, it could be due to reaching service limits, which may trigger the rate limiter to throttle your client. To determine if your client has reached service limits, you can enable logging and check if the service is returning errors with an HTTP status code 429. For more information on service limits, see the [Azure Monitor service limits documentation](https://learn.microsoft.com/azure/azure-monitor/service-limits#logs-ingestion-api).
+
+To enable client logging and to troubleshoot this issue further, see the instructions provided in the section titled [Enable client logging](#logging).
```
Comment (scottaddie) [inline sdk/monitor/azingest/TROUBLESHOOTING.md:76]:
```suggestion
To enable client logging and to troubleshoot this issue further, see the instructions provided in the section titled [Enable client logging](#enable-client-logging).
```

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md:25]: Don't have specific log flags for monitor ingestion, so the HTTP reuest and response events are the most relevant.

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: fixed in [d087998](https://github.com/Azure/azure-sdk-for-go/pull/21459/commits/d08799861214cf1cb2e7fdcf8243bd20210d2fef)

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 3 above)
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: Following up to check if a bug

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
+# Troubleshooting Azure Monitor Ingestion module
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Monitor Ingestion module for Go.
+
+## Table of contents
+
+* [General troubleshooting](#general-troubleshooting)
+    * [Error Handling](#error-handling)
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: fixed in [d087998](https://github.com/Azure/azure-sdk-for-go/pull/21459/commits/d08799861214cf1cb2e7fdcf8243bd20210d2fef)

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 5 above)
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: fixed in [d087998](https://github.com/Azure/azure-sdk-for-go/pull/21459/commits/d08799861214cf1cb2e7fdcf8243bd20210d2fef)

File: sdk/monitor/azingest/TROUBLESHOOTING.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    * [Troubleshooting missing logs](#troubleshooting-missing-logs)
+    * [Troubleshooting slow logs upload](#troubleshooting-slow-logs-upload)
+
+## General troubleshooting
+
+### Error Handling
+
+All methods which send HTTP requests return `*azcore.ResponseError` when these requests fail. `ResponseError` has error details and the raw response from Monitor Ingestion.
+
+### Logging
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: fixed in [d087998](https://github.com/Azure/azure-sdk-for-go/pull/21459/commits/d08799861214cf1cb2e7fdcf8243bd20210d2fef)

File: sdk/monitor/azingest/TROUBLESHOOTING.md (hunk 6 above)
Comment (gracewilcox) [reply sdk/monitor/azingest/TROUBLESHOOTING.md]: fixed in [d087998](https://github.com/Azure/azure-sdk-for-go/pull/21459/commits/d08799861214cf1cb2e7fdcf8243bd20210d2fef)

**Pull request merged**